docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1135 @@
+### 1.3. <a id="scope">Scope</a>
+This document describes the software architecture and software design requirements for The Schwarzenegger. This guide is mainly for developers, designers and software engineers that are or going to work on The Schwarzenegger. 
 
-## Product scope
-### Target user profile
+[&#8593; Return to Top](#developer-guide)
 
-{Describe the target user profile}
+## 2. <a id="setting-up">Setting Up</a>
 
-### Value proposition
+### 2.1. <a id="prerequisites">Prerequisites</a>
+1. JDK `11`.
+2. IntelliJ IDEA IDE.
 
-{Describe the value proposition: what problem does it solve?}
+### 2.2. <a id="setting-up-the-project-in-your-computer">Setting up the Project in Your Computer</a>
+1. Fork [this repository](https://github.com/AY2021S1-CS2113T-F11-1/tp), and clone the fork to your computer.
+2. Open IntelliJ (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project dialog first).
+3. Set up the correct JDK version for Gradle  
+   1. Click `Configure` > `Structure for New Projects` and then `Project Settings` > `Project` > `Project SDK`.
+   2. If `JDK 11` is listed in the drop down, select it. Otherwise, click `New…` and select the directory where you installed `JDK 11`.
+   3. Click `OK`.
+4. Click `Import Project`.
+5. Locate the `build.gradle` file and select it. Click `OK`.
+6. Click `Open as Project`.
+7. Click `OK` to accept the default settings if prompted. 
 
-## User Stories
+[&#8593; Return to Top](#developer-guide)
+
+## 3. <a id="design">Design</a>
+This section provides a high level overview of our application, The Schwarzenegger.
+### 3.1. <a id="architect">Architecture</a>
+
+![Architecture](pictures/zesong/Architecture.png)
+
+The image above explains the design of the application, The Schwarzenegger. 
+
+The main driver of the application is `Main: Duke`. It is responsible for mainly two phases:
+- At application launch
+    - This class will initialise the components in the correct sequence and is in charge of connecting them with each other.
+- At shut down
+    - This class will invoke cleanup method for the components when necessary.
+    
+In addition to that, the architecture of The Schwarzenegger is broken down into several packages, mainly the following: 
+- `Ui`: This class mainly handles the interactions with user of the application.
+- `Parser`: This class mainly handles the parsing and handling of user commands.
+- `Command`: This class handles the type of command.
+- `Profile`: This class manages the data of the user.
+- `Diet`: This class manages the diet recording sessions.
+- `Workout`: This class manages the data workout recording sessions.
+- `Storage`: This class reads data from and writes data back into a text file for future uses.
+
+[&#8593; Return to Top](#developer-guide)
+
+### 3.2. <a id="ui-component">Ui Component</a>
+![Ui Component](images/UI_Component.png)
```

Comment:
The link seems does not work as the pictures of Ui component does not shown, will it be better to update it?
![Screenshot 2020-10-29 at 12 31 54 PM](https://user-images.githubusercontent.com/69417796/97525844-bda5e000-19e2-11eb-9bc8-2e931b85328e.png)